types/time.go

```diff
@@ -1664,7 +1670,10 @@ func parseSingleTimeValue(unit string, format string, strictCheck bool) (int64,
 		}
 		dayCount := iv / int64(GoDurationDay/gotime.Second)
 		iv %= int64(GoDurationDay / gotime.Second)
-		return 0, 0, dayCount, iv*int64(gotime.Second) + dv*int64(gotime.Microsecond), err
+		if sign == int64(-1) {
+			return 0, 0, dayCount, -(iv*int64(gotime.Second) + dv*int64(gotime.Microsecond)), err
```

Comment:
Remove the negative conversation above, and using `iv*int64(gotime.Second) - dv*int64(gotime.Microsecond)` if negative here?